agent/config/config.go

```diff
@@ -238,7 +239,9 @@ func environmentConfig() Config {
 	// invalid parse
 	// Blank is not a warning; we have sane defaults
 	if err != io.EOF && err != nil {
-		seelog.Warnf("Invalid format for \"ECS_RESERVED_PORTS\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		msg := fmt.Sprintf("Invalid format for \"ECS_RESERVED_PORTS\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
```

Comment:
(multiple occurrences) I think you can just do `fmt.Errorf` and pass the resulting `error` to `seelog.Warn` without issue.  That will avoid the need to do `errors.New(string(msg))`.